### PR TITLE
MOTOLAB target FW support

### PR DESCRIPTION
**Pull request by shellixyz**

### CODE REVIEW

File: src/main/target/MOTOLAB/target.c [hunk 1]
```diff
@@ -23,14 +23,14 @@
...
-    { TIM3,  IO_TAG(PB1),  TIM_Channel_4, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR }, // PWM4  - PB1  - *TIM3_CH4, TIM1_CH3N, TIM8_CH3N
-    { TIM2,  IO_TAG(PA1),  TIM_Channel_2, 1, IOCFG_AF_PP,    GPIO_AF_1, TIM_USE_MC_MOTOR }, // PWM5  - PA1  - *TIM2_CH2, TIM15_CH1N
-    { TIM2,  IO_TAG(PA2),  TIM_Channel_3, 1, IOCFG_AF_PP,    GPIO_AF_1, TIM_USE_MC_MOTOR }, // PWM6  - PA2  - *TIM2_CH3, !TIM15_CH1
-    { TIM15, IO_TAG(PA3),  TIM_Channel_2, 1, IOCFG_AF_PP,    GPIO_AF_9, TIM_USE_MC_MOTOR | TIM_USE_MC_SERVO }, // PWM7  - PA3  - *TIM15_CH2, TIM2_CH4
-    { TIM1,  IO_TAG(PA8),  TIM_Channel_1, 1, IOCFG_AF_PP,    GPIO_AF_6, TIM_USE_MC_MOTOR | TIM_USE_MC_SERVO }, // PWM8  - PA8  - *TIM1_CH1, TIM4_ETR
+    { TIM3,  IO_TAG(PA4),  TIM_Channel_2, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR | TIM_USE_FW_MOTOR }, // PWM1  - PA4  - *TIM3_CH2
+    { TIM3,  IO_TAG(PA6),  TIM_Channel_1, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR | TIM_USE_FW_MOTOR }, // PWM2  - PA6  - *TIM3_CH1, TIM8_BKIN, TIM1_BKIN, TIM16_CH1
+    { TIM3,  IO_TAG(PB0),  TIM_Channel_3, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR | TIM_USE_FW_SERVO }, // PWM3  - PB0  - *TIM3_CH3, TIM1_CH2N, TIM8_CH2N
```
Comment (digitalentity) [inline src/main/target/MOTOLAB/target.c]: Using `TIM3` for both motor and servo is not recommended - this setup won't support OneShot and faster protocols.

Possible fix: use `TIM8_CH2N` and `TIM8_CN3N` for PB0 and PB1 pins. I'm not sure if timer N-channels will work though.

Comment (digitalentity) [reply src/main/target/MOTOLAB/target.c]: Another solution is to force motor/servo protocol to PWM@50HZ if using fixed-wing mixer - https://github.com/iNavFlight/inav/issues/2499#issuecomment-346973939

Comment (shellixyz) [reply src/main/target/MOTOLAB/target.c]: It is the same problem I encountered with the PIKOBLX target. This FC uses the same pins as PWM outputs. Using the timers with complementary channels did not work for me at the time but it is very possible I didn't do everything right.

I just checked the code and using complementary channels seems to be supported but it looks like the timer configuration output field needs to be ORed with the TIMER_OUTPUT_N_CHANNEL macro. See pwm_config.c:129. Can you confirm ?

I made the modifications to use `TIM8_CH2N` and `TIM8_CH3N` for PB0 and PB1 pins in this branch: https://github.com/shellixyz/inav/tree/MOTOLAB_FW_support_alt_timers

I will test this modification soon on the PIKOBLX FC soon.
@jcaamnes could you try on the MOTOLAB FC ?

Comment (shellixyz) [reply src/main/target/MOTOLAB/target.c]: I tested on PIKOBLX but it doesn't work. I get no signal on outputs 3 and 4.
https://github.com/shellixyz/inav/tree/PIKOBLX_alt_timers

**Comment by shellixyz** [comment]
I managed to make it work. I made a mistake in target.h but there was also a problem with pwm_output.c. Now different timers are used for PWM outputs 3/4 to use different timers for ESC and servos.

This PR now depends on #2512

**Comment by digitalentity** [comment]
This needs to be tested.

**Comment by digitalentity** [comment]
@shellixyz did you have a chance to test this PR?

**Comment by shellixyz** [comment]
@digitalentity I do not own this board but @jcaamnes should give us feedback in #2499 soon.

**Comment by digitalentity** [comment]
@shellixyz please give me a ping when this is ready for merge, I'd like to get all target-related changes in before I start to merge some of big stuff.

**Comment by shellixyz** [comment]
@digitalentity No feedback yet from @jcaamnes but in my opinion you can merge this safely. I tested the same timer modifications on the PIKOBLX target which uses the same MCU and output pins successfully (#2513).

**Comment by digitalentity** [comment]
Agreed, merging.

**Comment by jcaamnes** [comment]
i cannot get it to work. i flashed the hex file you sendt me @shellixyz. 
i connected motor to PWM1, servo to PWM3 and 4. set esc protocol to standard. When i arm, the motor is working, but i dont see any movements on the servos.

**Comment by jcaamnes** [comment]
@shellixyz sent me a new hex file with a fix that seems to work now :)

**Comment by mrduncan626** [comment]
Where can I find the motolabs hex file?  I don't see the updated hex file option in 1.9.1.

**Comment by shellixyz** [comment]
Looks like the MOTOLAB target is not built automatically you need to build it yourself. Here is the hex for 1.9-RC2:
[inav_1.9.0_MOTOLAB.hex.zip](https://github.com/iNavFlight/inav/files/1768764/inav_1.9.0_MOTOLAB.hex.zip)

**Comment by mrduncan626** [comment]
Thanks.  I'm not sure what you mean by "not built".  Can I just load this hex file locally in the iNav Firmware Flasher?

Also, is this a hex file for the MotoLabs F3 or F4  board?

**Comment by shellixyz** [comment]
Yes you can load locally and flash. Just remember 1.9 has not been released yet this is only a release candidate (RC). Only the MOTOLAB F3 is supported in iNav for the moment. A hex file has to be built for each target when a new version is released. Not automatically built means it is not built like the other targets are built so it is not available in the GUI to load with the "load firmware (online)" button. You need to build a hex file yourself if you want to use it.

**Comment by shellixyz** [comment]
@digitalentity Is there a reason for this target not to be included in the releases anymore since 1.3 ?

**Comment by digitalentity** [comment]
@shellixyz yes, it's my mistake 😄

**Pull request merged**